src/schema/schema_open.c

```diff
@@ -735,8 +736,21 @@ __wt_schema_open_layered(WT_SESSION_IMPL *session)
     WT_SAVE_DHANDLE(
       session, ret = __schema_open_layered_member(session, layered, layered->ingest_uri, true));
     WT_RET(ret);
+
     WT_SAVE_DHANDLE(
       session, ret = __schema_open_layered_member(session, layered, layered->stable_uri, false));
+    stable_deferred = ret == ENOENT && !S2C(session)->layered_table_manager.leader;
```

Comment:
Why we need to check this error? I think this is fragile. If we have something else return the same error, we are in trouble.

Reply:
We may not have any metadata for the stable table yet, so the schema open may fail when it tries to set the config for the stable table. I'm not sure how to design around this - `__conn_dhandle_config_set` explicitly converts the `WT_NOTFOUND` into an `ENOENT`.

Reply:
Can we pass in a parameter to indicate not found is OK in this case?

Reply:
That is possible but I think it's uglier than the current situation. I say that because of the amount of plumbing involved - the `ENOENT` ultimately comes from `__conn_dhandle_config_set`, but that's just one of a pile of function calls we make when opening a dhandle, so now we'd either need to plumb the flags to all of those calls, or to have some special `ENOENT` handling in the `err` clause of `__wt_conn_dhandle_open`.

One slight improvement would be to move the `ENOENT` logic to `__schema_open_layered_member` and include a comment? This would at least isolate the `ENOENT` source to `__wt_session_get_dhandle`.

Reply:
I agree it is better to move the logic to `__schema_open_layered_member `

Reply:
Fair enough, I've moved it there in `ccff628f1`.